firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/CustomModel.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.ml.modeldownloader;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.io.File;
+
+/**
+ * Used to store information about custom models that are being downloaded or are already downloaded
+ * on a device.
+ */
+public class CustomModel {
+  private String name;
+  private long downloadId;
+  private long fileSize;
+  private String modelHash;
+  private String localFilePath = "";
+
+  public @NonNull String getName() {
+    return name;
+  }
+
+  public @Nullable File getFile() {
```

Comment:
What can a model without a file be use for?

Reply:
It can be used to get the downloadId, which can be used to get the progress of the file's download status.

Reply:
Not sure I understand. How can I get a hold of an instance of `CustomModel` before the download has completed?
It looks like the only ways to get a `CustomModel` is to call download and wait for its task to complete, or list all "downloaded" models

Reply:
Good point, I forgot to add the addOnProgressListener to the getModel in the API design - it was discussed and planned (hence the download id). So, I added example. In the addOnProgressListener, the CustomModel is returned with the download id and can then be used to updated the progress bar, etc. Does this clarify?

Reply:
Thanks, it does. But it also reveals a potential design issue, namely, most of the getters have the following javadoc: "When models is not downloaded returns X, otherwise Y".

imo, the fact that the model has 2 modes of operation, indicates that type has low cohesion, and really represents 2 separate things:

* `ModelDownload` - contains info about the download that is in progress. Contains (modelName, downloadId), passed to onProgressListeners
* `CustomModel` - describes a fully downloaded model. Contains (name, file, size, hash), available only upon download completion.

This could reduce developer confusion, as they won't have to guess which state they are in(downloading vs downloaded). wdyt?

Reply:
We will actually have everything except the file during the progress listener. We have the model size and hash in advanced. So the difference is during "in progress" you'd use the download id to get download status information (which could be used to see if the download had started and % complete) and during "on success" you'd get the actual model file. We added the in progress functionality with download id so that a progress bar could be created and populated as desired.